File: go/0674E-最长连续递增序列.go
/*
 * @Author: SukiEva
 * @Date: 2021-12-06 09:23:13
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func findLengthOfLCIS(nums []int) int {
	ans, cnt := 1, 1
	for i := 1; i < len(nums); i++ {
		if nums[i] > nums[i-1] {
			cnt++
			ans = max(ans, cnt)
		} else {
			cnt = 1
		}
	}
	return ans
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: cpp/0674E-最长连续递增序列.cpp
/*
 * @Author: SukiEva
 * @Date: 2022-01-13 21:05:17
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int findLengthOfLCIS(vector<int>& nums) {
        vector<int> dp(nums.size(), 1);
        int ans = 1;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] > nums[i - 1]) dp[i] = dp[i - 1] + 1;
            ans = max(ans, dp[i]);
        }
        return ans;
    }
};